tasks/msi.py

```diff
@@ -43,13 +53,37 @@ def build(ctx, vstudio_root=None, arch="x64", major_version='7', python_runtimes
         env['SIGN_PFX'] = str(pfxfile)
         env['SIGN_PFX_PW'] = str(pfxpass)
 
+    env['AGENT_INSTALLER_OUTPUT_DIR'] = f'{BUILD_OUTPUT_DIR}'
+    env['NUGET_PACKAGES_DIR'] = f'{NUGET_PACKAGES_DIR}'
     print(f"arch is {arch}")
 
     cmd = ""
     configuration = "Release"
     if debug:
         configuration = "Debug"
 
+    # Copy source to build dir
+    # Hyper-v has a bug that causes the host's vmwp.exe to hold file locks indefinitely,
+    # preventing the build from overwriting output files. To work around this copy the
+    # source into the container, build on the container FS, then copy the output
+    # back to the mount.
```

Comment:
Why we do not have this problem for a regular Go build and our "legacy" Msi build?

Reply:
Good question, I don't know. For go, we already [copy the source out of the mount](https://github.com/DataDog/datadog-agent/blob/main/tasks/winbuildscripts/buildwin.bat#L7-L8) and into the go dir. I haven't been able to mount the source directly to the go dir for builds because of a separate [msys mount issue](https://datadoghq.atlassian.net/wiki/spaces/AW/pages/2567864326/Windows+Build+Environment#WindowsBuildEnvironment-Builderror:Notadirectoryortoomanylevelsofsymboliclinks).

Maybe it depends on the type of task, too. MSVC/CSC/MSBuild creates a lot of intermediate/output files in the source dir, go creates them in the `%WORK%`  dir which defaults to `%TEMP%` which is on the container FS. MSVC/CSC/MSBuild also have multiple output files whose projects may depend on each other (DLL -> EXE). But those things should apply to our custom action build, too, I think I've built that from mounts without issue.

This is consistently reproducible for me though, and never happens when built in the container filesystem.